statsd/statsd.go

```diff
@@ -43,6 +46,27 @@ func SetBackend(b Backend) {
 	currentBackend = b
 }
 
+// ErrUnknownBackend is returned when the statsd backend implementation is not known.
+var ErrUnknownBackend = fmt.Errorf("unknown statsd backend type")
+
+// NewBackend returns the appropriate Backend for the given implementation and host.
+func NewBackend(impl, addr, prefix string, tags ...string) (Backend, error) {
+	var err error
+	var b Backend
+	switch strings.ToLower(impl) {
+	case "datadog":
+		b, err = NewDatadogBackend(addr, prefix, tags)
+	case "log":
+		b = NewLogBackend(prefix, tags)
+	case "null":
+		b = NewNullBackend()
+	default:
+		return nil, errors.WithStack(ErrUnknownBackend)
+	}
+
+	return b, err
```

Comment:
The error in the default case is wrapped via `WithStack` (a good thing), seems like this should be done here as well. FWIW, `errors.Wrap(err, "...")` returns `nil`, when `err` is `nil`.